src/Raven.Server/ServerWide/SecretProtection.cs

```diff
@@ -851,15 +841,13 @@ public static void ValidateCertificateBeforeReplacement(X509Certificate2 certifi
             ValidateServerKeyUsages("ValidateCertificateBeforeReplacement", certificate, certificateValidationKeyUsages);
         }
 
-        internal static void ValidatePrivateKey(string source, string certificatePassword, byte[] rawData, out AsymmetricKeyEntry pk, SetupProgressAndResult progress = null)
+        internal static void ValidatePrivateKey(string source, string certificatePassword, byte[] rawData, out AsymmetricAlgorithm pk, SetupProgressAndResult progress = null)
         {
             pk = null;
-            foreach (string alias in GetAliases(certificatePassword, rawData, out var getKey))
-            {
-                pk = getKey(alias);
-                if (pk != null)
-                    break;
-            }
+            var certificate = CertificateLoaderUtil.CreateCertificate(rawData, certificatePassword, X509KeyStorageFlags.PersistKeySet);
+
+            // Get the private key.
+            pk = certificate.GetRSAPrivateKey();
```

Comment:
What is it isn't RSA?

Reply:
It is a separate issue, this one is a 1:1 transition